### PR TITLE
[Retry] Fix transitive framework import in opt build

### PR DESCRIPTION
**Pull request by qyang-nj**
**Retry merging PR #570, which solves issue #569.** The original PR got reverted because of issue #712.

### Original description
`apple_*_(xc)framework_import` provides `_SwiftInteropInfo` instead of `SwiftInfo`. In `apple_framework_packaging`, when [we merge the SwiftInfo](https://github.com/bazel-ios/rules_ios/blob/b1499fc08a7e4b7a7ba238fb4c9b089e6648ecbd/rules/framework.bzl#L901), the info from the framework import get lost there.

To fix the problem, we can add `swift_clang_module_aspect` to the `transitive_deps` attribute of `apple_framework_packaging`, which converts `_SwiftInteropInfo` to `SwiftInfo`. `swift_library` does this for `deps` as well.

@mattrobmattrob @thiagohmcruz

### CODE REVIEW

File: rules/import_middleman.bzl
Comment (qyang-nj) [inline rules/import_middleman.bzl:275]: This custom merging logic caused a problematic `modulemap` is generated. I'm not sure why we are merging `CcInfo` differently depending on whether VFS is used, but all CI jobs are passed.

**Comment by mattrobmattrob** [comment]
Running an integration test in our repo now. Do you mind doing the same when you have time, @thiagohmcruz?

**Review by mattrobmattrob — APPROVED** [review]
LGTM in our project. Let's let the Square/Block folks give it a test in their codebase before we merge though. 🙏

File: rules/internal/objc_provider_utils.bzl
Comment (qyang-nj) [inline rules/internal/objc_provider_utils.bzl:58]: Actually this is not used anywhere else.

**Review by jszumski — APPROVED** [review]
This worked for Cash 👍

**Comment by mattrobmattrob** [comment]
We can merge this and revert if Block folks have any issues. Thanks for fixing the previous problem, @qyang-nj!

**Comment by thiagohmcruz** [comment]
> We can merge this and revert if Block folks have any issues. Thanks for fixing the previous problem, @qyang-nj!

+1 thx for the PR @qyang-nj, sorry about the delay I'm intermittently available this week, will try to run some tests on this sometime this week

**Review by jerrymarino — COMMENTED** [review]
Thanks for the PR - left a comment inline and will be able to followup on this at a later day if necessary!

File: rules/import_middleman.bzl
Comment (thiagohmcruz) [reply rules/import_middleman.bzl:275]: We should be able to though right? If we can't rely on CI being 🟢 here then I think we should look into it and make CI start checking for whatever is necessary?

Comment (qyang-nj) [reply rules/import_middleman.bzl:275]: Yeah, I have the repro example in the issue #569 and the repro code [here](https://github.com/qyang-nj/BazelPlayground/tree/main/Issue0). I thought about adding it to CI tests, but realized we don't have any tests run under `--compilation_mode=opt`. Maybe we can add that.

Comment (jerrymarino) [reply rules/import_middleman.bzl:275]: SGTM! Yeah it'd be nice to improve the coverage for the things we can, I can dig more into this early next week.

File: rules/internal/objc_provider_utils.bzl
Comment (jerrymarino) [reply rules/internal/objc_provider_utils.bzl:58]: Overall this is "internal" but this overall utility module clearly has many useful functions none the less - we should clarify API contracts with this code.

**Pull request merged**
